### PR TITLE
Tracing Replace: Delete trace and trigger copy

### PR DESCRIPTION
**Pull request by diego-lon**
# Tracing Replace: Delete trace and trigger copy

This PR adds the code necessary to handle the deletion of the trace enriched and to trigger the copy of the csv file from `interop-tracing-replacement` bucket to `interop-tracing` bucket

## Tracing Enriched Data Handler
- Implements `processReplacementUploadMessage` that delete the tracing on database based on messages from queue `replacement-upload`
- Sends message to queue `enricher-state` to notify the deletion with `isReplacing: true`

## Tracing State Updater
- Consumes from `enricher-state` the message and triggers the api `triggerCopy` when message contains boolean `isReplacing`

## Tracing Operations
- Exposes api `/tracings/:tracingId/triggerCopy`
- Implements `bucketService` that creates path from tracing received and runs a `CopyObjectCommand` to copy the csv from `interop-tracing-replacement` bucket to `interop-tracing` bucket

### CODE REVIEW

File: packages/tracing-enriched-data-handler/test/init-db.sql
Comment (eduwardo) [inline packages/tracing-enriched-data-handler/test/init-db.sql:1]: Should have the same db sql of others microservices? @galales

**Review by eduwardo — APPROVED** [review]

File: packages/tracing-enriched-data-handler/src/services/db/dbService.ts [hunk 1]
```diff
@@ -52,17 +52,24 @@ export function dbServiceBuilder(db: DB) {
...
-          `Error inserting trace for tracingId: ${tracingId}: Details: ${error}`,
+        throw insertTracesError(
+          `Error inserting traces for tracingId: ${tracingId}. Details: ${error}`,
         );
       }
     },
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    async deleteTracing(_tracingId: string) {
+
+    async deleteTraces(tracingId: string) {
```
Comment (galales) [inline packages/tracing-enriched-data-handler/src/services/db/dbService.ts]: return type should be in the signature

File: packages/tracing-enriched-data-handler/src/services/db/dbService.ts [hunk 2]
```diff
@@ -52,17 +52,24 @@ export function dbServiceBuilder(db: DB) {
...
       try {
-        return Promise.resolve([{}]);
+        const deleteTracesQuery = `
+          DELETE FROM traces.traces
+          WHERE tracing_id = $1
+          RETURNING id
```
Comment (galales) [inline packages/tracing-enriched-data-handler/src/services/db/dbService.ts]: is it required to return the id?

File: packages/tracing-enriched-data-handler/test/init-db.sql
Comment (galales) [reply packages/tracing-enriched-data-handler/test/init-db.sql:1]: confirm that traces should be in a different schema

File: packages/tracing-state-updater/src/services/operationsService.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ export const operationsServiceBuilder = (
             state: data.state,
           },
           {
-            headers: { ...correlationIdToHeader(uuidv4()) },
+            headers: { ...correlationIdToHeader(uuidv4()) }, // Temporary value to valorize with correlationId implementation
```
Comment (galales) [inline packages/tracing-state-updater/src/services/operationsService.ts:32]: is this addressed in some other PR/task?

**Review by galales — APPROVED** [review]

File: packages/tracing-state-updater/src/services/operationsService.ts (hunk 3 above)
Comment (diego-lon) [reply packages/tracing-state-updater/src/services/operationsService.ts:32]: Resolved in #20

**Pull request merged**